### PR TITLE
Make the Ruleset buttons as the color of their mascots

### PR DESCRIPTION
**Pull request by Jinkku**
This pull request makes the ruleset buttons have their color as their mascot's color instead of having all of the buttons as the default light green color. This will not mess with custom rulesets as they will have the same color as osu!taiko

also I have put the color that was inside the `ToolbarRulesetTabButton.cs` to `OsuColour.cs` for in case the color can be reused if necessary

btw this is my first time coding in C# so my code might not be the best so feel free to edit it to your liking.

### CODE REVIEW

File: osu.Game/Overlays/Toolbar/ToolbarRulesetTabButton.cs [hunk 1]
```diff
@@ -53,14 +55,32 @@ public RulesetButton()
...
+                        }
+                        else if (tooltipMainText == "osu!catch")
+                        {
+                            rulesetcolour = colours.Blue;
+                        }
+                        else
+                        {
+                            rulesetcolour = colours.GreenLightest;
+                        }
+                        IconContainer.Colour = rulesetcolour;
```
Comment (bdach) [inline osu.Game/Overlays/Toolbar/ToolbarRulesetTabButton.cs:83]: Regardless of whether this is going to be accepted as an idea (I think it's a bad idea and will be way too noisy visually, but will ultimately defer to @peppy / @arflyte / whoever), this cannot be written this way. Magic string comparisons such as this are a last resort.

*If* we decide to do this, this should probably be moved to an abstract property on `Ruleset`.

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/Toolbar/ToolbarRulesetTabButton.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Overlays/Toolbar/ToolbarRulesetTabButton.cs:83]: Definitely.

File: osu.Game/Graphics/OsuColour.cs [hunk 2]
```diff
@@ -266,7 +266,7 @@ public static Color4 ForegroundTextColourFor(Color4 backgroundColour)
         public readonly Color4 Yellow = Color4Extensions.FromHex(@"ffcc22");
         public readonly Color4 YellowDark = Color4Extensions.FromHex(@"eeaa00");
         public readonly Color4 YellowDarker = Color4Extensions.FromHex(@"cc6600");
-
+        public readonly Color4 GreenLightest = Color4Extensions.FromHex(@"00ffaa");
```
Comment (peppy) [inline osu.Game/Graphics/OsuColour.cs:269]: Where did this come from?

**Comment by peppy** [comment]
I'm not strictly against this, but the colours would need to be adjusted to be legible. For instance, the osu!mania purpose is too dark.

File: osu.Game/Graphics/OsuColour.cs (hunk 2 above)
Comment (pxkidoescoding) [reply osu.Game/Graphics/OsuColour.cs:269]: I have moved the color that the tab buttons were using into that file

**Comment by peppy** [comment]
I mean, at very least this needs design side input, not just adjusting the colour arbitrarily.

**Comment by peppy** [comment]
Please open this as a discussion and we'll take it from there.

**Pull request closed without merging**